packages/module/src/Status/Status.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { Button, ButtonVariant, Flex, FlexItem, Popover, PopoverPosition, PopoverProps, Text, TextVariants, } from '@patternfly/react-core';
+import { createUseStyles } from 'react-jss';
+
+export const StatusVariant = {
+  link: 'link',
+  popover: 'popover',
+  plain: 'plain'
+} as const;
+
+export type StatusVariant = typeof StatusVariant[keyof typeof StatusVariant];
+
+export interface StatusProps extends React.PropsWithChildren {
+  /** Status label text */
+  label?: string;
+  /** Description to be displayed under the label */
+  description?: string;
+  /** If true, only displays icon */
+  iconOnly?: boolean;
+  /** Variant of the status component to be displayed */
+  variant?: StatusVariant;
+  /** Status icon */
+  icon?: React.ReactElement;
+  /** Custom OUIA ID */
+  ouiaId?: string | number;
+  /** Props for the optional popover */
+  popoverProps?: PopoverProps,
+  /** Optional link variant onClick callback */
+  onClick?: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+}
+
+const useStyles = createUseStyles({
+  icon: {
+    margin: 0
+  },
+  statusLabel: {
+    lineHeight: 'var(--pf-v5-global--LineHeight--sm)',
+  },
+  statusDescription: {
+    color: 'var(--pf-v5-c-content--small--Color)',
+  } 
+})
+
+export const Status: React.FC<StatusProps> = ({ variant = StatusVariant.plain, label, children, iconOnly, icon, ouiaId = 'Status', popoverProps, onClick, description, ...props }: StatusProps) => {
+  const classes = useStyles();
+
+  const statusBody = (
+    <Flex title={label} alignItems={{ default: 'alignItemsCenter' }} {...props}>
+      { icon && (
+        <FlexItem className={classes.icon}>
+          {React.cloneElement(icon, { className: clsx('pf-v5-u-mr-md', icon?.props?.className), title: label, 'data-ouia-component-id': `${ouiaId}-icon` })}
```

Comment:
I would recommend that the icon be wrapped in a [PF Icon component](https://www.patternfly.org/components/icon) and use the `status` prop on that component to control color rather than hardcoding a color in the passed icon.